### DISCUSSION TITLE
textcat doesn't work

### DISCUSSION THREAD

**Discussion by SergioMarreroMarrero** [explosion/spaCy#8124] [Help: Coding & Implementations] [answered]
Hi.

I am trying to run a simple texcat model with mutually exclusive categories. I have 3 categories: A, B, C

I have a _train.jsonl_ and _dev.jsonl_:
```jsonl
{"text": "This is an example", "cats": {"A": 1, "B": 0, "C": 0}}
{"text": "This is other example", "cats": {"A": 0, "B": 0, "C": 1}}
...
```

To build the binary format I apply this function (where the input data have the mentioned format (jsonl)):

```python
def convert(lang: str, input_path: Path, output_path: Path):
    nlp = spacy.blank(lang)
    db = DocBin()
    actions = []
    for line in srsly.read_jsonl(input_path):
        doc = nlp.make_doc(line["text"])
        doc.cats = line["cats"]
        db.add(doc)
    db.to_disk(output_path)

```

Finally, as a config file for training I am using [the config quickstart](https://spacy.io/usage/training#quickstart) with the options:
- Spanish
- textcat
- exclusive
- cpu
- efficiency

When I execute the train command...the process seems to start but eventually nothing happen:

![image](https://user-images.githubusercontent.com/18600509/118567708-2fbaf800-b76e-11eb-9ceb-909053bac214.png)

Can anyone help me? 

Thank you in advance.

**Comment by polm** [comment, marked as answer]
To clarify, does the training just exit by itself? That would be really weird, but it could happen with an out of memory error. 

If it doesn't just exit by itself, how long was it stuck before you stopped it?

How much training data do you have? Are you able to train if you reduce your data to a small number of examples, like 1000?

**Reply by SergioMarreroMarrero** [reply]
Helo @polm 

Thank you so much. Yes, it was a memory error. I expected a Memory Error message, and because that I didn't think on that could be the cause.

Thank so much again.

Sergio

**Reply by polm** [reply]
No worries, it's definitely confusing when it doesn't leave a message, and if I hadn't seen it before I wouldn't understand it. Unfortunately the operating system has the ability to unilaterally kill processes that use too much memory so there's not much we can do to improve this.
